proxy/envoy/ingress.go

```diff
@@ -137,14 +147,27 @@ func generateIngress(mesh *proxyconfig.ProxyMeshConfig, tls *model.TLSSecret, ce
 	}
 
 	config := buildConfig(listeners, nil, mesh)
+
+	h := sha256.New()
+	hashed := false
```

Comment:
We need to write the hash in two cases for Ingress: 1) when `tls` is not nil, which defines the cert/key for external traffic and 2) when `AuthPolicy` is `MUTUAL_TLS` and this takes care of certs used for traffic between proxies.